editoast/src/views/train_schedule.rs

```diff
@@ -358,40 +358,38 @@ async fn simulation(
         })
         .await?;
 
-    Ok(Json(
-        train_simulation(
-            &mut db_pool.get().await?,
-            valkey_client,
-            core_client,
-            train_schedule,
-            &infra,
-            electrical_profile_set_id,
-        )
-        .await?
-        .0,
-    ))
-}
-
-/// Compute simulation of a train schedule
-pub async fn train_simulation(
-    conn: &mut DbConnection,
-    valkey_client: Arc<ValkeyClient>,
-    core: Arc<CoreClient>,
-    train_schedule: TrainSchedule,
-    infra: &Infra,
-    electrical_profile_set_id: Option<i64>,
-) -> Result<(SimulationResponse, PathfindingResult)> {
-    Ok(train_simulation_batch(
-        conn,
+    // Compute simulation of a train schedule
+    let (simulation, _) = train_simulation_batch(
+        &mut db_pool.get().await?,
         valkey_client,
-        core,
+        core_client,
         &[train_schedule],
-        infra,
+        &infra,
         electrical_profile_set_id,
     )
     .await?
     .pop()
-    .unwrap())
+    .unwrap();
+
+    Ok(Json(simulation))
+}
+
+async fn retrieve_rolling_stocks_from_train_schedules(
+    conn: &mut DbConnection,
+    train_schedules: &[TrainSchedule],
+) -> Result<HashMap<String, RollingStockModel>> {
+    let rolling_stocks_ids = train_schedules
+        .iter()
+        .map::<String, _>(|t| t.rolling_stock_name.clone());
+    let (rolling_stocks, _): (Vec<_>, _) =
```

Comment:
Better use `retrieve_batch_or_fail` and deal with the error when ids are missing. If this situation can't happen, `retrieve_batch_unchecked` can be used with a comment explaining why it's safe to do so.

Reply:
We **MUST NOT** return an error if a rolling stock is not found here. See the new comments, doing so introduced a huge regression.

Reply:
Indeed. But then we have a documentation problem: we should at least add a comment above the instruction and document the function explicating its behavior. We can't just silently ignore a list of failures (albeit local ones) without "explaining why it's safe to do so".

@Wadjetz sorry about the misdirection, could you add these comments? You can also use `retrieve_batch_unchecked` to avoid computing the list of missing IDs. This will also simplify the `let` clause.